catalog/internal/catalog/rhec_catalog.go

```diff
@@ -0,0 +1,331 @@
+package catalog
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+
+	"github.com/golang/glog"
+	model "github.com/kubeflow/model-registry/catalog/pkg/openapi"
+	graphQL "github.com/shurcooL/graphql"
```

Comment:
Why this library? Any specific pros? How about something like https://github.com/Khan/genqlient for type safety and code generation? Have we looked at others at https://graphql.org/community/tools-and-libraries/?tags=go